### PR TITLE
fix: Save before computing aggregates

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation

This addresses a bug in the aggregator:  The aggregator would collect data for one SNS, but then compute aggregates before adding the data to the cache.  As a result it took unnecessarily long for new data to appear.

# Changes
* Move up the point at which the collected data is added to the cache.

# Tests
This has been deployed to testnets on multiple releases of the feature branch.

### CODE REVIEW

**Comment by bitdivine** [comment]
Svelte test fails aftre not changing frontend??

File: rs/sns_aggregator/src/state.rs [hunk 1]
```diff
@@ -92,6 +92,16 @@ impl State {
         let prefix = "/v1";
         let root_canister_id = convert_canister_id!(upstream_data.canister_ids.root_canister_id);
         let root_canister_str = root_canister_id.to_string();
+        // Add this to the list of values from upstream
+        STATE.with(|state| {
+            state
+                .stable
+                .borrow()
+                .sns_cache
```
Comment (lmuntaner) [inline rs/sns_aggregator/src/state.rs:100]: Not related to the change, but to the code. How does it know that when you do `sns_cache` or `upstream_data` is about a specific SNS and not another?

**Review by lmuntaner — APPROVED** [review]
Just on question about the code, not about the change.

Already approving

File: rs/sns_aggregator/src/state.rs (hunk 1 above)
Comment (bitdivine) [reply rs/sns_aggregator/src/state.rs:100]: I am not quite sre I understand the question.  `sns_cache` contains the data for all upstream sns's (eventually).  The `upstream_data` is  a map keyed by root canister ID; that is how it stores data for multiple SNSs.  Is that what you meant?  I have to close the conversation in order to merge the PR but am happy to follow up if needed.

Comment (lmuntaner) [reply rs/sns_aggregator/src/state.rs:100]: That answers my question, thanks!

**Pull request merged**